metaci/api/views/robot.py

```diff
@@ -0,0 +1,33 @@
+from rest_framework.renderers import JSONRenderer, BrowsableAPIRenderer
+from rest_framework import viewsets
+from django_filters.rest_framework import DjangoFilterBackend
+from metaci.api.renderers import SimpleCSVRenderer
+from metaci.api.serializers.robot import RobotTestResultSerializer
+from metaci.testresults.models import TestResult
+from metaci.testresults.filters import RobotResultFilter
+from rest_framework.filters import OrderingFilter
+
+
+class RobotTestResultViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    A viewset for viewing robot framework test results
+    """
+
+    serializer_class = RobotTestResultSerializer
+    renderer_classes = [BrowsableAPIRenderer, JSONRenderer, SimpleCSVRenderer]
+    filterset_class = RobotResultFilter
+
+    def get_queryset(self):
+        queryset = TestResult.objects.filter(method__testclass__test_type="Robot")
+
+        # FIXME: I'd like the user to be able to specify a period
+        # of time such as "last two weeks" or "this month"...
+        # I also need to
```

Comment:
I think this is probably essential to avoid trying to process too much data in production if the filters are not very selective. I might be inclined to just support specifying an integer number of days if it's easier.